docs/adr/0003-graphql-layer-between-backend-frontend.md

```diff
@@ -0,0 +1,104 @@
+# GraphQL layer between backend and frontend
+
+Date: 2020-04-30
+
+## Related tickets (not available for public)
+
+SIG-2568
+
+## Disclaimer
+
+Since this proposal is quite big the original author (jpoppe) has taken the liberty make this
+proposal a bit more opiniated then it normally should be. (One of the reasons is that the original
+author has done this implementation and fine tuned it a couple of times)
+
+## Context
+
+Since our current backend only serves data through REST and there is no possibility to generate
+generate types/interfaces there is no automated way to validate data on the frontend which is coming
+from the backend.
+
+Besides of that there are too many inconstencies between different backend calls and the current
+backend API documentation is not always correct and/or up to date because it's written manually
+and not generated from code.
+
+Because of this our `React components` are over complicated since we need to mangle data all the
```

Comment:
'overcomplicated' instead of 'over complicated'
Also, we don't mangle data, we transform API output to a format that the API can consume and that leads to bloat and too much client-side logic.